packages/mocha-runner/src/mocha-runner-factories.ts

```diff
@@ -9,15 +9,20 @@ import * as pluginTokens from './plugin-tokens';
 
 export const createMochaTestRunner = createMochaTestRunnerFactory();
 
-export function createMochaTestRunnerFactory(namespace: typeof INSTRUMENTER_CONSTANTS.NAMESPACE | '__stryker2__' = INSTRUMENTER_CONSTANTS.NAMESPACE) {
-  createMochaTestRunner.inject = tokens(commonTokens.injector);
-  function createMochaTestRunner(injector: Injector<PluginContext>): MochaTestRunner {
+export function createMochaTestRunnerFactory(
+  namespace: typeof INSTRUMENTER_CONSTANTS.NAMESPACE | '__stryker2__' = INSTRUMENTER_CONSTANTS.NAMESPACE
+): {
+  (injector: Injector<PluginContext>): MochaTestRunner;
+  inject: ['$injector'];
+} {
+  mochaTestRunner.inject = tokens(commonTokens.injector);
+  function mochaTestRunner(injector: Injector<PluginContext>): MochaTestRunner {
     return injector
       .provideClass(pluginTokens.loader, MochaOptionsLoader)
       .provideClass(pluginTokens.mochaAdapter, MochaAdapter)
       .provideClass(pluginTokens.directoryRequireCache, DirectoryRequireCache)
       .provideValue(pluginTokens.globalNamespace, namespace)
       .injectClass(MochaTestRunner);
   }
-  return createMochaTestRunner;
+  return mochaTestRunner;
```

Comment:
Why this name change? A `mochaTestRunner` is not returned, instead, a factory function is returned to create a mocha test runner with, hence the name was `createMochaTestRunner` :).

Reply:
In line 10, you export function as `createMochaTestRunner`, which triggers no-shadow in line 27(?) thats why i had to change the name. And since function is internal (inside other function), it shouldnt break anything.

Reply:
Renamed to `mochaTestRunnerFactory`